src/System.Net.Http/tests/StressTests/HttpStress/Program.cs

```diff
@@ -636,35 +616,48 @@ private static (string, string) GetGetQueryParameters(string contentSource, Clie
         return (queryString.ToString(), expectedString.ToString());
     }
 
-    /// <summary>Client context containing information pertaining to a single worker.</summary>
-    private sealed class ClientContext
+    /// <summary>Client context containing information pertaining to a single request.</summary>
+    private sealed class RequestContext
     {
         private readonly Random _random;
 
-        public ClientContext(HttpClient httpClient, int taskNum, int seed)
+        public RequestContext(HttpClient httpClient, Random random, int taskNum, double cancellationProbability)
         {
-            _random = new Random(Combine(seed, taskNum)); // derived from global seed and worker number
+            _random = random;
             TaskNum = taskNum;
             HttpClient = httpClient;
+            CancellationToken =
+                (GetRandomBoolean(cancellationProbability))
+                ? CreateCancellationTokenWithRandomizedCancellationDelay()
+                : CancellationToken.None;
 
-            // deterministic hashing copied from System.Runtime.Hashing
-            int Combine(int h1, int h2)
+            CancellationToken CreateCancellationTokenWithRandomizedCancellationDelay(int maxDelayMs = 5)
             {
-                uint rol5 = ((uint)h1 << 5) | ((uint)h1 >> 27);
-                return ((int)rol5 + h1) ^ h2;
+                var delay = TimeSpan.FromMilliseconds(GetRandomInt(maxDelayMs));
```

Comment:
On Windows at least there's going to be no difference between 1/2/3/4/5ms.  They will all end up getting rounded up to around 10-15ms.